scylla-cql/src/frame/request/mod.rs

```diff
@@ -48,3 +48,195 @@ pub trait Request {
 pub trait DeserializableRequest: Request + Sized {
     fn deserialize(buf: &mut &[u8]) -> Result<Self, ParseError>;
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{borrow::Cow, ops::Deref};
+
+    use bytes::Bytes;
+
+    use crate::{
+        frame::{
+            request::{
+                batch::{Batch, BatchStatement, BatchType},
+                execute::Execute,
+                query::{Query, QueryParameters},
+                DeserializableRequest, Request,
+            },
+            types::{self, LegacyConsistency, SerialConsistency},
+            value::SerializedValues,
+        },
+        Consistency,
+    };
+
+    #[test]
+    fn request_ser_de_identity() {
+        // Query
+        let contents = Cow::Borrowed("SELECT host_id from system.peers");
+        let parameters = QueryParameters {
+            consistency: Consistency::All,
+            serial_consistency: Some(SerialConsistency::Serial),
+            timestamp: None,
+            page_size: Some(323),
+            paging_state: Some(vec![2, 1, 3, 7].into()),
+            values: {
+                let mut vals = SerializedValues::new();
+                vals.add_value(&2137).unwrap();
+                Cow::Owned(vals)
+            },
+        };
+        let query = Query {
+            contents,
+            parameters,
+        };
+
+        {
+            let mut buf = Vec::new();
+            query.serialize(&mut buf).unwrap();
+
+            let query_deserialized = Query::deserialize(&mut &buf[..]).unwrap();
+            assert_eq!(&query_deserialized.contents, &query.contents);
+            assert_eq!(&query_deserialized.parameters, &query.parameters);
```

Comment:
Why not `assert_eq!(query_deserialized, query);`?

Reply:
This would require `PartialEq`.

Reply:
I've derived `PartialEq` under `test` cfg attr.